src/commonTest/kotlin/com/charleskorn/kaml/YamlWritingTest.kt

```diff
@@ -200,6 +200,69 @@ class YamlWritingTest : DescribeSpec({
                         """.trimMargin()
                 }
             }
+
+            context("serializing a string with the value of an integer using SingleLineStringStyle.PlainExceptAmbiguous") {
```

Comment:
Could you please add some test cases for when `singleLineStringStyle = SingleLineStringStyle.PlainExceptAmbiguous` but the value isn't considered ambiguous? eg. an ordinary string, a value like `1.2.3` that looks numeric but isn't etc.

Also, could you please add some test cases for encoding non-string values with `singleLineStringStyle = SingleLineStringStyle.PlainExceptAmbiguous` to ensure they're not quoted?